openupgrade_scripts/scripts/sale_stock/15.0.1.0/pre-migration.py

```diff
@@ -0,0 +1,24 @@
+from openupgradelib import openupgrade
+
+
+def _rename_field_product_packaging_to_product_packaging_id(env):
+    openupgrade.logged_query(
+        env.cr,
+        """
+        ALTER TABLE sale_order_line
+        ADD COLUMN IF NOT EXISTS product_packaging_id integer
+        """,
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE sale_order_line
+        SET product_packaging_id = product_packaging
+        WHERE product_packaging IS NOT NULL
+        """,
+    )
```

Comment:
I think that realize in a pre-migration.py script of ``sale`` module the following pseudo code could be simplier. Don't you think ? 

```
if is_installed("sale_stock"):
    rename("product_packaging", "product_packaging_id")
```

Reply:
@legalsylvain Your idea is excellent, however as you stated in the Sale module, it will be migrated first. Because product_packaging_id has already been generated in this scenario, renaming will result in an error.